tests/unit/models/gpflow/test_models.py

```diff
@@ -432,46 +596,336 @@ def test_gaussian_process_regression_trajectory_sampler_has_correct_samples(
+    x = gpflow.utilities.to_default_float(
+        tf.constant(np.arange(1, 24).reshape(-1, 1) / 8.0)
+    )  # shape: [23, 1]
+    y = fnc_2sin_x_over_3(x)
+
+    model5 = GaussianProcessRegression(gpr_model(x[:5, :], y[:5, :]))
+
+    additional_data = Dataset(tf.reshape(x[5:, :], [3, 6, -1]), tf.reshape(y[5:, :], [3, 6, -1]))
+
+    query_points = tf.concat([0.5 * x, 2.0 * x], 0)  # shape: [46, 1]
+
+    predj_mean5, predj_cov5 = model5.conditional_predict_joint(query_points, additional_data)
+    pred_mean5, pred_var5 = model5.conditional_predict_f(query_points, additional_data)
+    predy_mean5, predy_var5 = model5.conditional_predict_y(query_points, additional_data)
+
+    for i in range(3):
+        xi = tf.concat([x[:5, :], additional_data.query_points[i, ...]], axis=0)
+        yi = tf.concat([y[:5, :], additional_data.observations[i, ...]], axis=0)
+
+        modeli = GaussianProcessRegression(gpr_model(xi, yi))
+        predj_meani, predj_covi = modeli.predict_joint(query_points)
+        pred_meani, pred_vari = modeli.predict(query_points)
+        predy_meani, predy_vari = modeli.predict_y(query_points)
+
+        np.testing.assert_allclose(predj_mean5[i, ...], predj_meani, atol=1e-5)
+        np.testing.assert_allclose(pred_meani, pred_mean5[i, ...], atol=1e-5)
+        np.testing.assert_allclose(pred_vari, pred_var5[i, ...], atol=1e-5)
+        np.testing.assert_allclose(predj_covi, predj_cov5[i, ...], atol=1e-5)
+        np.testing.assert_allclose(predy_vari, predy_var5[i, ...], atol=1e-5)
+        np.testing.assert_allclose(predy_vari, predy_var5[i, ...], atol=1e-5)
+
+
+def test_gaussian_process_regression_conditional_predict_f_sample() -> None:
+    x = gpflow.utilities.to_default_float(
+        tf.constant(np.arange(1, 24).reshape(-1, 1) / 8.0)
+    )  # shape: [23, 1]
+    y = fnc_2sin_x_over_3(x)
+
+    model5 = GaussianProcessRegression(gpr_model(x[:5, :], y[:5, :]))
+    additional_data = Dataset(tf.reshape(x[5:, :], [3, 6, -1]), tf.reshape(y[5:, :], [3, 6, -1]))
+    query_points = tf.concat([0.5 * x, 2.0 * x], 0)  # shape: [46, 1]
+    samples = model5.conditional_predict_f_sample(query_points, additional_data, num_samples=100000)
+    npt.assert_array_equal([3, 100000, 46, 1], samples.shape)
+
+    for i in range(3):
+        xi = tf.concat([x[:5, :], additional_data.query_points[i, ...]], axis=0)
+        yi = tf.concat([y[:5, :], additional_data.observations[i, ...]], axis=0)
+
+        modeli = GaussianProcessRegression(gpr_model(xi, yi))
+        predj_meani, predj_covi = modeli.predict_joint(query_points)
+        sample_mean = tf.reduce_mean(samples[i], axis=0)
+        sample_cov = tfp.stats.covariance(samples[i, :, :, 0], sample_axis=0)
+        np.testing.assert_allclose(sample_mean, predj_meani, atol=1e-2, rtol=1e-2)
+        np.testing.assert_allclose(sample_cov, predj_covi[0], atol=1e-2, rtol=1e-2)
+
+
+def test_sparse_gaussian_process_regression_raises_for_invalid_init() -> None:
```

Comment:
from here until line 920 is the new part, I have mostly reused tests from gpr and svgp, I didn't include some of the svgp tests on inducing points that required a two-output model - @henrymoss do you think it would be important to add those?

Reply:
No leave the two outputs for SVGP